lib/widgets/payment_request_dialog.dart

```diff
@@ -20,48 +27,237 @@ class PaymentRequestDialog extends StatefulWidget {
   PaymentRequestDialog(this.context, this.accountBloc, this.invoice);
 
   @override
-  State<StatefulWidget> createState() {    
+  State<StatefulWidget> createState() {
     return PaymentRequestDialogState();
   }
 }
 
-class PaymentRequestDialogState extends State<PaymentRequestDialog> {
+class PaymentRequestDialogState extends State<PaymentRequestDialog>
+    with SingleTickerProviderStateMixin {
+
   final _formKey = GlobalKey<FormState>();
   TextEditingController _invoiceAmountController = new TextEditingController();
 
+  AnimationController controller;
+  Animation<double> opacityAnimation;
+  Animation<double> borderAnimation;
+  Animation<RelativeRect> transitionAnimation;
+  Animation<Color> colorAnimation;
+
+  PaymentRequestState _state;
+
+  Int64 _amountToPay;
+  String _amountToPayStr;
+
+  bool _inProgress = false;
+  bool _isInit = false;
+
   @override
-  void initState() {    
+  void initState() {
     super.initState();
-    _invoiceAmountController.addListener((){
+    widget.accountBloc.accountStream.listen((acc) {
+      if (acc.paymentRequestInProgress != null && acc.paymentRequestInProgress.isNotEmpty) {
+        _inProgress = true;
+      } else if (acc.paymentRequestInProgress == null || acc.paymentRequestInProgress.isEmpty){
+        _inProgress = false;
+      }
+    });
+    _listenPaymentsResults();
+    setState(() {
```

Comment:
We shouldn't use `setState` within `initState` method. You can do the assignment without `setState`.